doc/compiler/il/notes_mainline_outOfLine_GlRegDeps.md

```diff
@@ -0,0 +1,97 @@
+n190n       ==>iand
+n582n     ificmpne --> block_57 BBStart at n553n ()                                           [0x7fec3b0ba5a0] bci=[1,30,-] rc=0 vc=0 vn=- li=- udi=- nc=3 flg=0x20
+n580n       iand                                                                              [0x7fec3b0ba500] bci=[1,30,-] rc=1 vc=0 vn=- li=- udi=- nc=2
+n578n         iloadi  <isClassFlags>[#287  Shadow +36] [flags 0x603 0x0 ]                     [0x7fec3b0ba460] bci=[1,30,-] rc=1 vc=0 vn=- li=- udi=- nc=1
+n577n           aloadi  <componentClass>[#282  Shadow +88] [flags 0x10607 0x0 ]               [0x7fec3b0ba410] bci=[1,30,-] rc=1 vc=0 vn=- li=- udi=- nc=1
+n576n             aloadi  <vft-symbol>[#288  Shadow] [flags 0x18607 0x0 ]                     [0x7fec3b0ba3c0] bci=[1,30,-] rc=1 vc=0 vn=- li=- udi=- nc=1
+n506n               ==>aRegLoad
+n579n         iconst 1024                                                                     [0x7fec3b0ba4b0] bci=[1,30,-] rc=1 vc=0 vn=- li=- udi=- nc=0
+n581n       iconst 0                                                                          [0x7fec3b0ba550] bci=[1,30,-] rc=1 vc=0 vn=- li=- udi=- nc=0
+n583n       GlRegDeps ()                                                                      [0x7fec3b0ba5f0] bci=[1,30,-] rc=1 vc=0 vn=- li=- udi=- nc=3 flg=0x20
+n506n         ==>aRegLoad
+n505n         ==>iRegLoad
+n584n         PassThrough rdx                                                                 [0x7fec3b0ba640] bci=[1,29,-] rc=1 vc=0 vn=- li=- udi=- nc=1
+n190n           ==>iand
+n572n     BBEnd </block_33>                                                                   [0x7fec3b0ba280] bci=[1,30,-] rc=0 vc=0 vn=- li=- udi=- nc=0
+```
+
+The `xRegStore` serves the purpose of anchoring those children that are honored and ensuring that they get evaluated before the `GlRegDeps` gets encountered. However, the `xRegStore` does not make sure that the value is in that real register at the point of `xRegStore`.
+
+`xRegStore` is not responsible for putting the value into the register. It is not actually going to do anything at that point other than evaluates the child and associates it with a virtual register. The actual moving off that value into the right real register is dictated by `GlRegDeps` at the end of block. The local register allocator in the code gen happens backwards. It walks backwards through the instructions. It will see `GlRegDeps` first, or the result of `GlRegDeps` first. It associates the virtual register with the real register the first time.
+
+The purpose of `xRegStore` and `GlRegDeps` is only to specify which value should be in which real register at those points while transitioning from one block to another. It does not mandate the value has to be in this real register at this point. It ensures the right values are in the right real registers at those transition points.
+
```

Comment:
The `xRegStore` does not mandate the value has to be in the right real register at that point but the `GlRegDeps` is in fact at the point where we transition from one block to another and so it is one of the points where the values have to be in the right registers.